src/sdl2/render.rs

```diff
@@ -1099,31 +1168,51 @@ impl<T: RenderTarget> Canvas<T> {
     }
 
     /// Sets the clip rectangle for rendering on the specified target.
-    ///
-    /// If the rectangle is `None`, clipping will be disabled.
     #[doc(alias = "SDL_RenderSetClipRect")]
-    pub fn set_clip_rect<R: Into<Option<Rect>>>(&mut self, rect: R) {
-        let rect = rect.into();
-        // as_ref is important because we need rect to live until the end of the FFI call, but map_or consumes an Option<T>
-        let ptr = rect.as_ref().map_or(ptr::null(), |rect| rect.raw());
-        let ret = unsafe { sys::SDL_RenderSetClipRect(self.context.raw, ptr) };
+    pub fn set_clip_rect<R>(&mut self, arg: R)
+    where
+        R: Into<ClippingRect>,
+    {
+        let arg: ClippingRect = arg.into();
+        let ret = match arg {
+            ClippingRect::Some(r) => unsafe {
+                sys::SDL_RenderSetClipRect(self.context.raw, r.raw())
+            },
+            ClippingRect::Zero => {
+                let r = sys::SDL_Rect {
+                    x: 0,
+                    y: 0,
+                    w: 0,
+                    h: 0,
+                };
+                let r: *const sys::SDL_Rect = &r;
+                unsafe { sys::SDL_RenderSetClipRect(self.context.raw, r) }
+            }
+            ClippingRect::None => unsafe {
+                sys::SDL_RenderSetClipRect(self.context.raw, ptr::null())
+            },
+        };
         if ret != 0 {
             panic!("Could not set clip rect: {}", get_error())
         }
     }
 
     /// Gets the clip rectangle for the current target.
-    ///
-    /// Returns `None` if clipping is disabled.
     #[doc(alias = "SDL_RenderGetClipRect")]
-    pub fn clip_rect(&self) -> Option<Rect> {
+    pub fn clip_rect(&self) -> ClippingRect {
+        let clip_enabled = unsafe { sys::SDL_RenderIsClipEnabled(self.context.raw) };
+
+        if let sys::SDL_bool::SDL_FALSE = clip_enabled {
```

Comment:
binding is unnecessary here, just do `if sys::SDL_bool::SDL_FALSE == clip_enabled {`

Reply:
fails to compile:

```bash
error[E0308]: mismatched types
    --> src/sdl2/render.rs:1205:12
     |
1205 |         if sys::SDL_bool::SDL_FALSE = clip_enabled {
     |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `bool`, found `()`
     |
help: consider adding `let`
     |
1205 |         if let sys::SDL_bool::SDL_FALSE = clip_enabled {
     |            +++

```